DietPi-Software | ADS-B Feeder: Harden uninstall and fix beta tags

dirkhh opened the pull request:
This addresses the concern https://github.com/MichaIng/DietPi/pull/6587#issuecomment-1743756614 about the ADS-B Feeder app sometimes issuing a docker prune which potentially removes data that we didn't own. The corresponding change to the upstream code was submitted as well: https://github.com/dirkhh/adsb-feeder-image/commit/0136d44a2c771066d4f28eeb641765d9f699ea34

This also adapts to a recent change in the tag semantic in the ADS-B Feeder image upstream and cleans up the services handling.

dirkhh committed:
correctly show beta tags

The tag format upstream was extended - this tracks that change.

Signed-off-by: Dirk Hohndel <dirk@hohndel.org>

dirkhh committed:
ADS-B Feeder: improve service integration

Don't install the unused update timer - we don't use that as an app.
Remove the adsb-docker service on uninstall.
Don't prune the docker images - that might remove images we don't own.

Signed-off-by: Dirk Hohndel <dirk@hohndel.org>

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -12015,11 +12015,13 @@ _EOF_
 			# determine the version
 			local ADSB_FEEDER_DATE_COMPONENT=$(git log -20 --date='format:%y%m%d' --format='%ad' | uniq -c | mawk '{print $2"."$1;exit}')
-			local ADSB_FEEDER_TAG_COMPONENT=$(git describe --match 'v[0-9]*' | cut -d- -f1)
+			local ADSB_FEEDER_TAG_COMPONENT=$(git describe --match "v[0-9]*" --long | sed "s/-[0-9]*-g[0-9a-f]*//")
```
Consequent single-quote usage if there are no variables to expend:
```suggestion
			local ADSB_FEEDER_TAG_COMPONENT=$(git describe --match 'v[0-9]*' --long | sed 's/-[0-9]*-g[0-9a-f]*//')
```

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -12814,11 +12816,9 @@ If no WireGuard (auto)start is included, but you require it, please do the follo
 			Remove_Service adsb-feeder-update
 			Remove_Service adsb-setup
-			Remove_Service adsb-update
-			command -v docker &> /dev/null && G_EXEC docker image prune -a -f
```
Does this mean the image may remain on the system? Probably we can search for a matching container and image, if they have a consistent name pattern, like we do on Portainer uninstall:
```sh
			# Check if Docker is still installed
			if [[ -d '/mnt/dietpi_userdata/docker-data' ]]
			then
				# Remove ADS-B Feeder container & image
				local container=$(docker container ls -a | mawk '/adsb-feeder( |$)/{print $1;exit}')
				[[ $container ]] && G_EXEC docker container rm -f "$container"
				local image=$(docker image ls -a | mawk '/adsb-feeder( |$)/{print $3;exit}')
				[[ $image ]] && G_EXEC docker image rm "$image"
			fi
```

dirkhh replied:
I tried to figure out if that was worth doing and ended up giving up. The names of the images are... fairly inconsistent. I could add a list of them here, but that seemed very noisy and very specific.
I think it would make more sense to do this upstream in the adsb-feeder-image repo and have a helper function that could be called here. I'll have to play with that.

MichaIng replied:
Ah right, makes sense with the large number of containers. Images can be given a tag right? Probably that can be used to make them identifiable. Or otherwise some kind of state file?

dirkhh committed:
Update dietpi/dietpi-software

applying suggestion via the web interface - I know I will regret this...

Co-authored-by: MichaIng <micha@dietpi.com>

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -12015,11 +12015,13 @@ _EOF_
 			# remove the service that isn't needed for an app install on DietPi and install the rest
 			G_EXEC cd /tmp/adsb-feeder/src/modules/adsb-feeder/filesystem/root
 			G_EXEC rm ./usr/lib/systemd/system/adsb-bootstrap.service
+			G_EXEC rm ./usr/lib/systemd/system/adsb-update.service
+			G_EXEC rm ./usr/lib/systemd/system/adsb-update.timer
```
Shall we add a [patch](https://github.com/MichaIng/DietPi/blob/dev/.update/patches) to remove this service+timer on next DietPi update, if present? Otherwise it would currently remain on systems which did install ADS-B Feeder with DietPi v8.22 already.

dirkhh replied:
yes, that makes sense.

dirkhh committed:
remove unwanted service files

The names should be specific enough for this approach.

Signed-off-by: Dirk Hohndel <dirk@hohndel.org>

dirkhh committed:
run upstream cleanup script

This should allow us to clean up the various docker images that might have been
installed.

Signed-off-by: Dirk Hohndel <dirk@hohndel.org>

dirkhh commented on .update/patches:
```diff
@@ -1480,6 +1480,10 @@ Patch_8_23()
+	# ADS-B Feeder used to install two service files that aren't needed
+	# the names should be specific enough to make an unconditional force remove acceptable
+	G_EXEC rm -f /etc/systemd/system/adsb-update.service
+	G_EXEC rm -f /etc/systemd/system/adsb-update.timer
```
I wasn't sure if this is the preferred way to do this - feel free to point me at a better approach

MichaIng replied:
Let's be conservative and only touch things if it was really installed via `dietpi-software`. And disable those, if enabled.
```sh
	# ADS-B Feeder used to install two service files that aren't needed: https://github.com/MichaIng/DietPi/pull/6661
	if [[ -f '/boot/dietpi/.installed' ]] && grep -q '^[[:blank:]]*aSOFTWARE_INSTALL_STATE\[141\]=2' /boot/dietpi/.installed
	then
		systemctl -q is-enabled adsb-update.timer && G_EXEC systemctl --no-reload disable --now adsb-update.timer
		[[ -f '/etc/systemd/system/adsb-update.timer' ]] && G_EXEC rm /etc/systemd/system/adsb-update.timer
		systemctl -q is-enabled adsb-update && G_EXEC systemctl --no-reload disable --now adsb-update
		[[ -f '/etc/systemd/system/adsb-update.service' ]] && G_EXEC rm /etc/systemd/system/adsb-update.service
	fi
```

dirkhh replied:
I didn't know that `disable` also supports a `--now` flag... I thought this one was asymmetric for some strange reason... and of course I have no idea what you are actually doing with the grep on `/boot/dietpi/.installed` ... something for me to study a bit, I guess 🙃

MichaIng replied:
The grep checks whether the software title with ID 141 is installed (`=2`).

dirkhh commented on dietpi/dietpi-software:
```diff
@@ -12819,6 +12819,7 @@ If no WireGuard (auto)start is included, but you require it, please do the follo
 			Remove_Service adsb-docker
 			Remove_Service adsb-feeder-update
 			Remove_Service adsb-setup
+			[[ -f /opt/adsb/pre-uninstall-cleanup ]] && G_EXEC /opt/adsb/pre-uninstall-cleanup
```
right now this script doesn't exist... but this seems like the easy way to make sure that once I have figured out a clean and consistent and easy to maintain way to do this for all images that the ADS-B Feeder might install now (and that can easily be kept current by upstream without having to make changes in DietPi)

MichaIng replied:
Sounds reasonable.

dirkhh committed:
improve the patch to remove services that might be still around

This is 100% written by @MichaIng - I'm just adding to the branch.

Signed-off-by: Dirk Hohndel <dirk@hohndel.org>

MichaIng committed:
v8.23

- CHANGELOG | ADS-B Feeder: The uninstall has been hardened to rule out the removal of unused Docker images which were not created by ADS-B Feeder. Furthermore, beta tags are now correctly shown in the version string. Many thanks to @andreagdipaolo for reporting a related issue and @dirkhh for implementing the enhancement: https://github.com/MichaIng/DietPi/pull/6587#issuecomment-1743744008

MichaIng committed:
v8.23

- DietPi-Patches | Remove superseded patch
